Add missing R recipes (#96)
posit-dev/connect-cookbook: tags/assign/index.qmd

## Python

### Prerequisites

Before executing, ensure you have the following information:

- The unique identifier (GUID) for the content item you want to tag. See [Content] recipes, or copy the GUID from the Info pane in the Connect Dashboard.

> - The unique identifier the tag you want to use. See the [Find tags](../find/index.qmd) recipe for details.

Author: In this recipe, the prerequisites differ between R and Python. I'm putting the Prerequisites section within the tab view. I'll add another subheading to delineate the actual workflow from the requirements.

```{.python}
from posit import connect
client = connect.Client()

content_guid = 'your-content-item-guid'
tag_id = "your-tag-id"

client.post(f"/v1/content/{content_guid}/tags", json={

>     'tag_id': tag_id,

Author: Python expected output is missing.

## Python

### Prerequisites

Before executing, ensuree you have the following information:

- The unique identifier (GUID) for the content item you want to tag. See [Content] recipes, or copy the GUID from the Info pane in the Connect Dashboard.

> - The identifier the tag you want to use. See the [Find tags](../find/index.qmd) recipe for details.

Author: Python and R workflows for tags differ significantly, so for some recipes, it made sense to move the "Prerequisites" section inside the language tabs.

### Code

```{.python}
from posit import connect
client = connect.Client()

content_guid = 'your-content-item-guid'

> tag_id = "your-tag-id"

Author: I made these changes thinking I'd be making things more consistent, but I don't think I am. I think we need to standardize how we refer to example GUIDs etc.

- The name or identifier of the tag you wish to find. See the [Find tags](../find/index.qmd) recipe for details.

### Code

The `set_content_tags()` function in `connectapi` can take multiple tag objects, and returns an object representing the updated content item.

```{.r}
library(connectapi)
client <- connect()

# Create an object representing the content you wish to update.
content_guid <- "154bd2af-e8fa-4aa4-aab8-dcef701f4af9"
target_content <- get_content(client, guid = content_guid)

# Get the tags from the Connect server, then specify the tag you wish to apply
# to the content.
tags <- get_tags(client)
target_tag <- tags$category$my_tag

target_content <- set_content_tags(content = target_content, tag = target_tag)
```

### Expected Output

```{.r}
> target_content <- set_content_tags(target_content, target_tag)
Posit Connect Tag Tree (target_content)
└── category
   └── my_tag

> target_content
Posit Connect Content: 
  Content GUID: 154bd2af-e8fa-4aa4-aab8-dcef701f4af9
  Content URL: https://connect.example.org/connect/#/apps/154bd2af-e8fa-4aa4-aab8-dcef701f4af9/
  Content Title: Public Content States

> content_item(client, guid = "154bd2af-e8fa-4aa4-aab8-dcef701f4af9")

Author: In the R code, I'm using a real GUID in the examples because the expected output includes real GUIDs.

### Prerequisites

> Before executing, ensuree you have the following information:

Reviewer: ```suggestion
Before executing, ensure you have the following information:
```

- The unique identifier (GUID) for the content item you want to tag. See [Content] recipes, or copy the GUID from the Info pane in the Connect Dashboard.

> - The identifier the tag you want to use. See the [Find tags](../find/index.qmd) recipe for details.

Reviewer: ```suggestion
- The identifier for the tag you want to use. Please look at the [Find tags](../find/index.qmd) recipe for details on obtaining the identifier.
```

> ### Code

Reviewer: Should this be... 
```suggestion
### Recipe
```

Author: I think we should probably revisit some of the template, because this is already under a heading `## Recipe` heading. It's like… we should decide what we mean when we refer to the "recipe": the document, the code, the workflow, the problem, etc.

Author: I'll try to address this in the template PR. Going to leave as-is here for now.
